jmespath-core/src/main/java/io/burt/jmespath/Adapter.java

```diff
@@ -82,6 +94,12 @@
   T getProperty(T value, String name);
 
   /**
+   * Returns all of the property names of the given object, or an empty list
+   * when the given value is not an object.
+   */
+  Collection<String> getPropertyNames(T value);
```

Comment:
Fair enough. The current comment says it should return a list, but the declaration says collection.